hpycc/connection.py

```diff
@@ -21,20 +21,45 @@
 from tempfile import TemporaryDirectory
 from time import sleep
 from warnings import warn
+import threading
+from urllib.parse import quote_plus
+from json import JSONDecodeError
+from simplejson.errors import JSONDecodeError as simpleJSONDecodeError
 
 from hpycc.utils.parsers import parse_wuid_from_failed_response, \
     parse_wuid_from_xml
 from hpycc import delete
 
+lock = threading.Lock()
+
+
+def _make_thorname_html(logical_file):
+    """
+    quote_plus in urllib is so badly named I made this function.
+    """
+    return quote_plus(logical_file)
+
 
 def check_ecl_cmd(cmd='ecl'):
```

Comment:
I really like this function, but again. Is it the scope of this pull request? I feel it should be a new issue

Reply:
Perhaps but function names were highlighted as a limitation  in the previous pull request, hence this addition. This was a bug encountered while developing this solution (it caused tests to fail) so was added here.